### PR TITLE
ARROW-10457: [CI] Fix Spark branch-3.0 integration tests

### PR DESCRIPTION
**Pull request by arw2019**
cc @BryanCutler

### CODE REVIEW

**Comment by arw2019** [comment]
@github-actions crossbow submit  test-conda-python-3.7-spark-branch-3.0

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-conda-python-3.7-spark-branch-3.0

File: ci/docker/conda-python-spark.dockerfile [hunk 1]
```diff
@@ -20,21 +20,15 @@ ARG arch=amd64
 ARG python=3.6
 FROM ${repo}:${arch}-conda-python-${python}
 
-ARG jdk=8
-ARG maven=3.5
+ARG spark=3.0.1
 
 # The Spark tests currently break with pandas >= 1.0
```
Comment (BryanCutler) [inline ci/docker/conda-python-spark.dockerfile:25]: This shouldn't be the case any more

**Review by BryanCutler — COMMENTED** [review]
Thanks for working on this @arw2019. My intention was to test pyarrow against spark branch-3.0 in addition to the current tests that also include arrow Java for the master branch. We still want to maintain that to be sure spark will be able to build with the latest arrow.

**Comment by arw2019** [comment]
> My intention was to test pyarrow against spark branch-3.0 in addition to the current tests that also include arrow Java for the master branch. We still want to maintain that to be sure spark will be able to build with the latest arrow.

Ok! In that case do we want to add another Docker build to the suite with spark branch-3.0 + pyarrow?

File: ci/docker/conda-python-spark.dockerfile (hunk 1 above)
Comment (arw2019) [reply ci/docker/conda-python-spark.dockerfile:26]: Ok! I reverted the changes here and bumped pandas to the latest release (1.2) and hopefully the build will go green

**Comment by arw2019** [comment]
@github-actions crossbow submit test-conda-python-3.7-spark-branch-3.0

**Comment by arw2019** [comment]
@github-actions crossbow submit test-conda-python-3.7-spark-branch-3.0

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-conda-python-3.7-spark-branch-3.0

**Comment by arw2019** [comment]
This build is currently failing because of a problem with the Java implementation:
```
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project arrow-plasma: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test failed: Plugin org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3 or one of its dependencies could not be resolved: Failed to collect dependencies at org.apache.maven.plugins:maven-surefire-plugin:jar:3.0.0-M3 -> org.apache.maven.surefire:maven-surefire-common:jar:3.0.0-M3: Failed to read artifact descriptor for org.apache.maven.surefire:maven-surefire-common:jar:3.0.0-M3: Could not transfer artifact org.apache.maven.surefire:maven-surefire-common:pom:3.0.0-M3 from/to central (https://repo.maven.apache.org/maven2): Connection reset -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <goals> -rf :arrow-plasma

```

**Comment by jorisvandenbossche** [comment]
@BryanCutler any idea about that failure?

**Comment by BryanCutler** [comment]
From a quick glance it looks like a connection problem with maven central, so lets try one more time

**Comment by BryanCutler** [comment]
@github-actions crossbow submit test-conda-python-3.7-spark-branch-3.0

**Comment by kszucs** [comment]
@github-actions crossbow submit test-conda-python-3.7-spark-branch-3.0

**Comment by kszucs** [comment]
We definitely should test against spark releases.

I executed the build against 3.0.1 but it fails with:
```
- max records in batch conf *** FAILED ***
  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 21.0 failed 1 times, most recent failure: Lost task 0.0 in stage 21.0 (TID 25, 5fc0f8cfe8d2, executor driver): java.lang.NoClassDefFoundError: Could not initialize class
 org.apache.spark.sql.util.ArrowUtils$
```

**Comment by kszucs** [comment]
```
Caused by: java.lang.RuntimeException: No DefaultAllocationManager found on classpath. Can't allocate Arrow buffers. Please consider adding arrow-memory-netty or arrow-memory-unsafe as a dependency.
```

This change was introduced by https://github.com/apache/arrow/commit/2092e18752a9c0494799493b12eb1830052217a2
Which is part of the previous release, so I assume a testing issue.

**Comment by BryanCutler** [comment]
@arw2019 sorry, I probably should have been more clear in the JIRA description, but there are more things needed to properly test against a Spark maintenance branch like branch-3.0. In the interest of time, I made another PR https://github.com/apache/arrow/pull/9210 with what I think needs to happen. I'll try out the tests there to hopefully get them passing soon.

**Comment by arw2019** [comment]
Closing in favour of #9210

**Pull request closed without merging**